src/algorithms/all_paths.c

```diff
@@ -81,12 +82,13 @@ size_t AllPaths
     unsigned int minLen,
     unsigned int maxLen,
     size_t *pathsCap,
-    Path **paths
+    Path **paths,
+    bool locked
```

Comment:
This argument is safe, but I admit I don't love it. The alternative is to pass the GraphContext through these functions instead of the Graph, which would be...effectively the same.